src/modules/dotnet.rs

```diff
@@ -46,13 +49,89 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         get_version_from_cli()?
     };
 
-    module.set_style(config.style);
-    module.create_segment("symbol", &config.symbol);
-    module.create_segment("version", &config.version.with_value(&version.0));
+    let current_tfm = find_current_tfm(&dotnet_files);
+
+    let formatter = if let Ok(formatter) = StringFormatter::new(config.format) {
+        formatter
+            .map_style(|variable| match variable {
+                "style" => Some(config.style.to_string()),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "symbol" => Some(config.symbol.to_string()),
+                "version" => Some(version.0.to_string()),
+                "tfm" => {
+                    if current_tfm.is_some() {
+                        Some(current_tfm.as_ref().unwrap().to_string())
+                    } else {
+                        None
+                    }
+                }
+                _ => None,
+            })
+    } else {
+        log::warn!("Error parsing format string in `dotnet.format`");
+        return None;
+    };
+
+    module.set_segments(formatter.parse(None));
+
+    module.get_prefix().set_value("");
+    module.get_suffix().set_value("");
 
     Some(module)
 }
 
+fn find_current_tfm<'a>(files: &[DotNetFile<'a>]) -> Option<String> {
+    let get_file_of_type = |t: FileType| files.iter().find(|f| f.file_type == t);
+
+    let relevant_file = get_file_of_type(FileType::ProjectFile).or_else(|| files.iter().next())?;
```

Comment:
The `or_else` doesn't seem to have a purpose here, given the match below will ignore anything which isn't a `ProjectFile`.